storage/postgres/plugin.go

```diff
@@ -183,3 +216,177 @@ func (p *PostgresBackend) DeletePluginById(ctx context.Context, id string) error
 
 	return nil
 }
+
+func (p *PostgresBackend) FindReviewById(ctx context.Context, db queryer, id string) (types.ReviewDto, error) {
+	query := fmt.Sprintf(`SELECT * FROM %s WHERE id = $1 LIMIT 1;`, REVIEWS_TABLE)
+
+	rows, err := db.Query(ctx, query, id)
+	if err != nil {
+		return types.ReviewDto{}, err
+	}
+
+	defer rows.Close()
+
+	review, err := pgx.CollectOneRow(rows, pgx.RowToStructByName[types.Review])
+	if err != nil {
+		return types.ReviewDto{}, err
+	}
+
+	ratings, err := p.FindRatingByPluginId(ctx, db, review.PluginId)
+	if err != nil {
+		return types.ReviewDto{}, err
+	}
+
+	var reviewDto types.ReviewDto
+
+	reviewDto.ID = review.ID
+	reviewDto.Address = review.Address
+	reviewDto.Comment = review.Comment
+	reviewDto.CreatedAt = review.CreatedAt
+	reviewDto.Rating = review.Rating
+	reviewDto.PluginId = review.PluginId
+	reviewDto.Ratings = ratings
+	return reviewDto, nil
+}
+
+func (p *PostgresBackend) FindReviews(ctx context.Context, pluginId string, skip int, take int, sort string) (types.ReviewsDto, error) {
+	if p.pool == nil {
+		return types.ReviewsDto{}, fmt.Errorf("database pool is nil")
+	}
+
+	allowedSortingColumns := map[string]bool{"created_at": true}
+	orderBy, orderDirection := common.GetSortingCondition(sort, allowedSortingColumns)
+
+	query := fmt.Sprintf(`
+		SELECT *, COUNT(*) OVER() AS total_count
+		FROM %s
+		WHERE plugin_id = $1
+		ORDER BY %s %s
+		LIMIT $2 OFFSET $3`, REVIEWS_TABLE, orderBy, orderDirection)
+
+	rows, err := p.pool.Query(ctx, query, pluginId, take, skip)
+	if err != nil {
+		return types.ReviewsDto{}, err
+	}
+
+	defer rows.Close()
+
+	var reviews []types.Review
+	var totalCount int
+
+	for rows.Next() {
+		var review types.Review
+
+		err := rows.Scan(
+			&review.ID,
+			&review.Address,
+			&review.Rating,
+			&review.Comment,
+			&review.CreatedAt,
+			&review.PluginId,
+			&totalCount,
+		)
+		if err != nil {
+			return types.ReviewsDto{}, err
+		}
+
+		reviews = append(reviews, review)
+	}
+
+	pluginsDto := types.ReviewsDto{
+		Reviews:    reviews,
+		TotalCount: totalCount,
+	}
+
+	return pluginsDto, nil
+}
+
+func (p *PostgresBackend) CreateReview(ctx context.Context, reviewDto types.ReviewCreateDto, pluginId string) (types.ReviewDto, error) {
+	if p.pool == nil {
+		return types.ReviewDto{}, fmt.Errorf("database pool is nil")
+	}
+
+	tx, err := p.pool.Begin(ctx)
+	if err != nil {
+		return types.ReviewDto{}, fmt.Errorf("failed to begin transaction: %w", err)
+	}
+
+	defer func() {
```

Comment:
there is a general WithTransaction function in postgres/db.go in [this PR](https://github.com/LimeChain/vultiserver-plugin/pull/73) which can potentially be used